### PR TITLE
Fixed #29448 -- Added a management command for creating management commands.

### PR DESCRIPTION
**Pull request by glasnt**
Includes refactoring the management template to allow for more than just app and project

### CODE REVIEW

**Comment by glasnt** [comment]
https://code.djangoproject.com/ticket/29448

File: django/core/management/templates.py [hunk 1]
```diff
@@ -304,7 +311,7 @@ def extract(self, filename):
         Extract the given file to a temporarily and return
         the path of the directory with the extracted content.
         """
-        prefix = 'django_%s_template_' % self.app_or_project
+        prefix = 'django_%s_template_' % self.tempalte_type
```
Comment (adamchainz) [inline django/core/management/templates.py]: typo

File: django/core/management/templates.py [hunk 2]
```diff
@@ -74,9 +76,14 @@ def handle(self, app_or_project, name, target=None, **options):
...
+            
             if not os.path.exists(top_dir):
-                raise CommandError("Destination directory '%s' does not "
-                                   "exist, please create it first." % top_dir)
+                # Creating recursive directories to store management commands is valid
+                if template_type == "command":
+                    os.makedirs(top_dir)
+                else: 
+                    raise CommandError("Destination directory '%s' does not "
+                                       "exist, please create it first." % top_dir)
```
Comment (adamchainz) [inline django/core/management/templates.py]: Shouldn't 'command' at least fail if the given app dir doesn't exist? Seems to me there should be a way of writing this code so 'command' isn't such a special case

File: django/core/management/templates.py [hunk 3]
```diff
@@ -20,14 +20,15 @@
...
-    Copy either a Django application layout template or a Django project
-    layout template into the specified directory.
+    Copy a Django template layout template into the specified directory.
+
+    Type of: app, project, (management) command
 
     :param style: A color style object (see django.core.management.color).
-    :param app_or_project: The string 'app' or 'project'.
-    :param name: The name of the application or project.
+    :param template_type: The string 'app', 'project', or 'command'.
```
Comment (adamchainz) [inline django/core/management/templates.py]: Perhaps alphabetizing the list would make the most sense, so "'app', 'command', or 'project'" ?

File: django/core/management/commands/startcommand.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django.core.management.templates import TemplateCommand
+import os
```
Comment (adamchainz) [inline django/core/management/commands/startcommand.py]: isort is going to ask you to rearrange this

File: django/core/management/commands/startcommand.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class Command(TemplateCommand):
+    help = (
+        "Creates a Django management command for the given command name  "
```
Comment (adamchainz) [inline django/core/management/commands/startcommand.py]: "with the given command name" ?

File: django/core/management/commands/startcommand.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        "in the management/commands folder of the specified app."
+    )
+
+    missing_args_message = "You must provide an application and command name"
```
Comment (adamchainz) [inline django/core/management/commands/startcommand.py]: "you must provide application and command names" ?

File: django/core/management/templates.py (hunk 2 above)
Comment (glasnt) [reply django/core/management/templates.py]: Ah! Yes, I can make the template dir include the `management/command` folder structure.

**Comment by glasnt** [comment]
Closing based on comments in the original issue. 

> ​[Django Extensions already implements a `create_command` command](https://django-extensions.readthedocs.io/en/latest/command_extensions.html), which sets up the folders and puts a command scaffolding in (​[that looks like this](https://github.com/django-extensions/django-extensions/blob/master/django_extensions/conf/command_template/management/commands/sample.py.tmpl)).

**Pull request closed without merging**